### PR TITLE
rewrite requestJob a bit

### PR DESCRIPTION
**Pull request by Graziel**
see #95 

ended up rewriting requestJob,

please double check :)

### CODE REVIEW

File: src/Model/Table/QueuedTasksTable.php [hunk 1]
```diff
@@ -240,51 +234,33 @@ public function requestJob(array $capabilities, $group = null) {
...
-		// Generate a list of already fetched ID's and a where clause for the update statement
-		$capTimeout = Hash::combine($capabilities, '{s}.name', '{s}.timeout');
-		foreach ($data as $item) {
-			$whereClause[] = '(id = ' . $item['id'] . ' AND (workerkey IS NULL OR fetched <= "' . date('Y-m-d H:i:s', time() - $capTimeout[$item['jobtype']]) . '"))';
-			if (!empty($item['fetched'])) {
-				$wasFetched[] = $item['id'];
-			}
+		if ($job->fetched) {
+			$job = $this->patchEntity($job, [
+				'failed' => $job->failed+1,
```
Comment (dereuromark) [inline src/Model/Table/QueuedTasksTable.php]: spaces around +

**Comment by dereuromark** [comment]
See failing tests https://travis-ci.org/dereuromark/cakephp-queue/jobs/138265525
But nice idea
Just needs some more confirmation here we dont break stuff or we need a major version bump.

**Comment by Graziel** [comment]
im getting ` ->Sry, the Retry Example Job failed. Try again.<-` in `QueueRetryExampleTask` the test passes but im not sure if the message is correct?

**Comment by dereuromark** [comment]
At the least your feature seems to break PHP5.4 compatability: 

    ......PHP Fatal error:  Class 'Cake\I18n\FrozenTime' not found in
    /home/travis/build/dereuromark/cakephp-queue/src/Model/Table/QueuedTasksTable.php on line 188

**Comment by Graziel** [comment]
yeah FrozenTime was available from cake 3.2 i think it was cause of this it was bumped to 5.5 ive already changed it to Time but still not sure about that message

**Comment by Graziel** [comment]
another bump

**Comment by dereuromark** [comment]
Wow, quite some cleanup here - so much more readable.
Thx!

**Pull request merged**